shank-macro-impl/src/macros/program_id.rs

```diff
@@ -40,17 +40,14 @@ impl TryFrom<&[ItemMacro]> for ProgramId {
             Err(ParseError::new_spanned(
                 &matches[0].0,
                 format!(
-                    "Found more than one program id candidate: {:?}",
-                    matches
-                        .iter()
-                        .map(|x| x.1.clone())
-                        .collect::<Vec<String>>()
+                    "Found more than one program id candidate: {:?}. Specify one with -p",
+                    matches.iter().map(|x| x.1.clone()).collect::<Vec<String>>()
                 ),
             ))
         } else if matches.is_empty() {
             Err(ParseError::new(
                 Span::call_site(),
-                "Could not find a `declare_id(\"<program-id>\")` invocation in the program",
+                "Could not find a `declare_id(\"<program-id>\")` invocation in the program. Override program address with -p",
```

Comment:
```suggestion
            "Could not find a `declare_id(\"<program-id>\")` invocation in the program. If this is intentional provide a program address via the -p argument instead"
```

As a result most test assertions need to be adapted

Reply:
Updated